components/style/context.rs

```diff
@@ -38,6 +38,10 @@ pub struct SharedStyleContext {
     /// Screen sized changed?
     pub screen_size_changed: bool,
 
+    /// Skip the root during traversal?
+    #[cfg(feature = "gecko")]
+    pub skip_root: bool,
```

Comment:
Well, I should stop reviewing without the gecko changes. Please add a note here than (as a followup) we can skip submitting the first job to the work queue (that would only contain the root) and send all the children instead.